### PR TITLE
Run Firestore Quickstart with Emulator Suite

### PR DESCRIPTION
**Pull request by jhuleatt**
Add instructions to show how to run the Firestore uppercaser quickstart locally with the [Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite)

### CODE REVIEW

File: quickstarts/uppercase-firestore/README.md [hunk 1]
```diff
@@ -1,19 +1,64 @@
...
 
 ## Introduction
 
 This sample app does two things:
 
-- Create messages in Cloud Firestore using a simple HTTPS request which is
+- Creates messages in Cloud Firestore using a simple HTTPS request which is
   handled by an HTTP function. Writing to Cloud Firestore is done using the
   Firebase Admin SDK.
 - When a message gets added in the Cloud Firestore, a function triggers and
```
Comment (egilmorez) [inline quickstarts/uppercase-firestore/README.md:13]: Suggest omitting "the". (just "in Cloud Firestore")

File: quickstarts/uppercase-firestore/README.md [hunk 2]
```diff
@@ -1,19 +1,64 @@
...
+
+The
+[Firebase Local Emulator Suite](https://firebase.google.com/docs/emulator-suite)
+allows you to build and test apps on your local machine instead of deploying to
+a Firebase project.
+
+1. Create a Firebase project in the
+   [Firebase Console](https://console.firebase.google.com)
+   > _Wondering why this step is needed?_ Even though the emulator will run this
+   > sample on your local machine, it needs to associate with a Firebase project
```
Comment (egilmorez) [inline quickstarts/uppercase-firestore/README.md:26]: This sounds a little too anthropomorphized as/is . . . though it introduces passive voice, I'd suggest "needs to be associated with" -- or maybe just "needs to interact with?"

File: quickstarts/uppercase-firestore/README.md [hunk 3]
```diff
@@ -1,19 +1,64 @@
...
+   > to retrieve some configuration values.
+1. [Set up or update the Firebase CLI](https://firebase.google.com/docs/cli#setup_update_cli)
+1. Run `firebase emulators:start`
+1. Open the Emulator Suite UI
+   1. Look in the output of the `firebase emulators:start` command for the url
```
Comment (egilmorez) [inline quickstarts/uppercase-firestore/README.md:31]: Should URL be capped?  If it's a literal, use backticks.

File: quickstarts/uppercase-firestore/README.md [hunk 4]
```diff
@@ -1,19 +1,64 @@
...
+      of the Emulator Suite UI. It defaults to
+      [localhost:4000](http://localhost:4000), but may be hosted on a different
+      port on your machine.
+   1. Enter that URL in your browser to open the UI.
+1. Trigger the functions
+   1. Look in the output of the `firebase emulators:start` command for the url
```
Comment (egilmorez) [inline quickstarts/uppercase-firestore/README.md:37]: Same comment about "url" here.

File: quickstarts/uppercase-firestore/README.md (hunk 1 above)
Comment (jhuleatt) [reply quickstarts/uppercase-firestore/README.md]: removed

File: quickstarts/uppercase-firestore/README.md (hunk 2 above)
Comment (jhuleatt) [reply quickstarts/uppercase-firestore/README.md]: I changed to "interact with" - LGTY?

File: quickstarts/uppercase-firestore/README.md (hunk 3 above)
Comment (jhuleatt) [reply quickstarts/uppercase-firestore/README.md]: Switched to capitalized URL

File: quickstarts/uppercase-firestore/README.md (hunk 4 above)
Comment (jhuleatt) [reply quickstarts/uppercase-firestore/README.md]: Switched to capitalized URL

**Comment by jhuleatt** [comment]
> This is great! Do you possibly see replacing the RTDB quickstart for CF3 with this?

Yes, I'd like to add it to as many samples as possible. I wanted to focus on just one first since it will mostly just be copy & paste between samples

**Comment by markarndt** [comment]
LGTM. @abeisgoat, can you check the UI log viewer step in these instructions? I think it's correct for the beta launch and will still be correct for all of the fast-follow changes too....?

**Pull request merged**